apps/src/code-studio/scriptAnnouncementsRedux.js

```diff
@@ -7,6 +10,13 @@ export const addAnnouncement = (notice, details, link, type) => ({
   announcementType: type,
 });
 
+export const announcementShape = PropTypes.shape({
+  notice: PropTypes.string.isRequired,
+  details: PropTypes.string.isRequired,
+  link: PropTypes.string.isRequired,
+  type: PropTypes.oneOf(Object.values(NotificationType)).isRequired,
+});
```

Comment:
It feels weird to put this in redux (as redux is generally meant to be React agnostic, and PropTypes are not), however I wanted to share this shape somewhere, and creating an entirely new file felt silly. Still not sure what I think about this.

Reply:
I don't mind as long as we're not doing something crazy like pushing React components into our store.